src/main/scala/zio/s3/Live.scala

```diff
@@ -105,47 +108,58 @@ final class Live(unsafeClient: S3AsyncClient) extends S3.Service {
       .flatMap(publisher =>
         execute(
           _.putObject(
-            PutObjectRequest
-              .builder()
-              .bucket(bucketName)
-              .contentLength(contentLength)
-              .contentType(contentType)
-              .key(key)
-              .metadata(metadata.asJava)
-              .build(),
+            {
+              val builder = PutObjectRequest
+                .builder()
+                .bucket(bucketName)
+                .contentLength(contentLength)
+                .key(key)
+                .metadata(options.metadata.asJava)
+                .acl(options.cannedAcl)
+              options.contentType
+                .fold(builder)(builder.contentType)
+                .build()
+            },
             AsyncRequestBody.fromPublisher(publisher)
           )
         )
       )
       .unit
 
-  // only for file which are bigger than 5 Mb
   def multipartUpload[R <: zio.Has[_]: Tag](
     bucketName: String,
     key: String,
-    contentType: String,
     content: ZStream[R, Throwable, Byte],
-    metadata: Map[String, String] = Map.empty
+    options: MultipartUploadOptions = MultipartUploadOptions()
   ): ZIO[R, S3Exception, Unit] =
     for {
       uploadId <- execute(
-                    _.createMultipartUpload(
-                      CreateMultipartUploadRequest
+                    _.createMultipartUpload {
+                      val builder = CreateMultipartUploadRequest
```

Comment:
you can remove this `val builder`

Reply:
how? The content type is optional, that's why there is a builder that is then optionally updated with a content type.

Reply:
oups my bad all good 👍